Tests/Shared.Specs/TaskAssertionSpecs.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+using System.Threading.Tasks;
+using FluentAssertions.Extensions;
+using Xunit;
+using Xunit.Sdk;
+
+namespace FluentAssertions.Specs
+{
+    public class TaskAssertionSpecs
```

Comment:
You are testing here `Func<Task>` not `Task`.
So this tests should be placed in the file for the other Func tests.

Reply:
Which one do you mean?

Reply:
AsyncFunctionExceptionAssertionSpecs.cs

Reply:
I suspected that one, but that one acts on exceptions, not tasks.

Reply:
Then I recommend `AsyncMethodAssertionSpecs.cs` here and `AsyncFunctionAssertionSpecs.cs` below.
Or - following your implementation classes `GenericAsyncFunctionAssertionsSpecs.cs` and `NonGenericAsyncFunctionAssertionsSpecs.cs`.

BTW: I myself prefer positive logic. This is why I do not that like "NonGeneric".